.github/workflows/gradle.yml

```diff
@@ -37,5 +35,43 @@ jobs:
       run: sudo apt-get install omlib-modelica-3.0
     - name: Grant execute permission for gradlew
       run: chmod +x gradlew
-    - name: Test with Gradle
-      run: ./gradlew test -i
+    - name: Build with Gradle
+      run: ./gradlew build -i
+  release:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out ${{ github.ref }}
+      uses: actions/checkout@v2
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'adopt'
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Build Jars with Gradle
+      run: ./gradlew shadowClientJar shadowServerJar
+    - name: Add version from build.gradle to output
+      id: version
+      run: echo "::set-output name=version::$(./gradlew getVersion -q)"
+    - name: Print Version
+      run: echo ${{ steps.version.outputs.version }}
+    - name: Release
+      uses: softprops/action-gh-release@v1
+      with:
+        files: |
+          ./build/libs/thm.mope.lsp-${{ steps.version.outputs.version }}-Server.jar
+          ./build/libs/thm.mope.lsp-${{ steps.version.outputs.version }}-ConsoleClient.jar
+        token: ${{ secrets.GITHUB_TOKEN }} #Provided by GithubActions
+        target_commitish: ${{github.sha}} #Get latest commit on this branch
+        draft: false
+        prerelease: true
+        tag_name: v${{ steps.version.outputs.version }}
```

Comment:
This works for your setup, but is not fully correct. If you name your next tag `v0.1.0` instead of `0.1.0`, this would fail. Instead, you should use `github.ref` and remove the trailing `refs/tag/` from the full ref, e.g. by using the  parameter expansion of the shell like this: `${GITHUB_REF##*/}`. In this case, all trailing patterns of the form `*/` found in the variable `$GITHUB_REF` are removed, leaving just the tag name (or any other remaining part of the ref that is not followed by a slash).